src/generator.cs

```diff
@@ -4653,6 +4684,39 @@ void GenerateProperty (Type type, PropertyInfo pi, List<string> instance_fields_
 			indent--;
 			print ("}\n");
 			return;			
+		} else if (minfo.has_inner_wrap_attribute) {
+			// If propperty getter or setter has its own WrapAttribute we let the user do whatever their heart desires
+			if (pi.CanRead) {
+				PrintAttributes (pi, platform: true);
+				PrintAttributes (pi.GetGetMethod (), platform: true, preserve: true, advice: true);
+				print ("get {");
+				indent++;
+
+				print ($"return {minfo.wpmi.WrapGetter};");
+
+				indent--;
+				print ("}");
+			}
+			if (pi.CanWrite) {
+				var setter = pi.GetSetMethod ();
+				var not_implemented_attr = AttributeManager.GetCustomAttribute<NotImplementedAttribute> (setter);
+
+				PrintAttributes (pi, platform: true);
+				PrintAttributes (setter, platform: true, preserve: true, advice: true, notImplemented: true);
+				print ("set {");
+				indent++;
+
+				if (not_implemented_attr != null)
+					print ($"throw new NotImplementedException (@\"{not_implemented_attr?.Message}\");");
```

Comment:
This will generate `throw new NotImplementedException ("");` (and not `throw new NotImplemented ();` from `[NotImplemented]`. Maybe check for a null message like this: https://github.com/xamarin/xamarin-macios/blob/bec116c273ab119c648522a28c3f650881ed4d9e/src/generator.cs#L4738?